### ISSUE TITLE
Regular college society (University)

### ISSUE THREAD

**Issue by SEANSOKEN** [Hexenstag-Team/WH-Hexenstag-Dev#64]
Regular college society (University)
As a society where you can join/be sent to as a 16 to 25 yearold.
Has up to 6 ranks, first being the entry rank, last 2 being profesor and decan.
You start by selecting what field to study in  and start at rank 1 which is entry rank.

You live at the college from cash you have or that has been sent by your parents. You can demand more cash via decision when you run out or more regularly if you are greedy. Your parents can refuse to pay and if you stall for too long on the degree ranks, your dad/father in law can come in and beat you up.
The cash can be used for paying college fees (not paying will kick you out), partying, attempt of bribing the profesors, wasting on fancy stuff and just living your days, extending time counter on passing a degree.
Getting kicked out of the college for not successfully passing degree in time, you get an event to go back for a fee. And if you refuse, your kin that sent you to college (if someone sent you to it) will get the event to resend you back.
Failing exams and getting sent back over or just having bad time can give depression or depressed.

You gain society points by events and passing exams(missions). If you are a lauzy student, you can try and bribe for passing exams or offer sex to the professor for a pass.
Ability to pass and gain society points are heavily influenced by your traits and stats. Having diligent or genious gives you heavy advantage and assures you that you will pass unless you are very unlucky or have an asshole professor that keeps failing you.
Gaining enough society points gives you an ability to rank up in the society which is finishing a degree. 
There is a time counter, in what time you have to rank up in the society otherwise you will be kicked out of the college and you will leave with the degree you had your society rank at. If you only had the first rank, you utterly failed college and have no degree.
Getting the last degree (fourth society rank) grants you an option to finish college fully or continue studying and trying to get professor rank (next society rank) and stay in the college society as a regular member with no time counter to kick you after you become a professor. Becoming a professor (society rank 5) still has a time counter in what you need to gain enough society points as for degrees before.
If you have a hard time with a professor or score, you can ask for a commity to step in. Depending on your stats there is a chance they let you pass and schold or fire the professor, or they realize you are an idiot and fail you properly.
Finishing the degree gives you that academic trait strength depending on the society rank you had. First rank you get nothing, drop out. Second rank gives you tier 1 of the academic degree of the field you selected, third rank gives you tier 2 and fourth rank gives you tier 3. You can also become a physician(also 3 tiers). 
Getting the fifth rank, the professor rank, just gives passive rank bonuses and the sixth rank is the decan that has even more passive rank bonuses.

As a profesor you get the events to grade students and fail or pass them. You can be a cunt or pretty leniant. All the bribing events and seduction you get as a student now you get as a professor, so you can take bribes and pass students if they offered sex.
Professors that have lustful trait can try and initiate sexual affair with a student, players get this option too.

Getting found out about bribes sexual affairs gets you both expeled and labeled adultress.

**Comment by HobbitDerrick** [comment]
My main issue with this idea is that it's a society that the player should almost never join. (It doesn't seem like the kind of thing that a ruler would be doing.) Now, I probably like the AI more than most people and appreciate the intention to make the AI have fun - but the AI hardly ever enjoys this kind of content. ;)

**Comment by SEANSOKEN** [comment]
Why not? I hate it that you cant send your kid when they are landed. It makes no sense, it is the Empire university, YOU want your kid to finish it. And if they stay and become a professor or a decan, and they inherit you, well you have a pretty strong position.
Also true, in a lot of cases you will have to die early to play this. Tho that is why Otto suggested that you can join the acadamy at any age.

**Comment by HobbitDerrick** [comment]
A ruler is already doing a full time job that's a kind of hybrid between a small business owner, a public official, and a military officer. To casually throw full-time studies on top of that like it doesn't mean anything is abandoning believability. I'm sympathetic to rewriting the current buggy and unimpressive university events, but it's not necessary to have a society to get those events right. Turning the University titles into playable positions is also possible, and would achieve almost everything else you've described here - often without the need for custom events, because of pre-existing court interactions.

Let me turn the question around, though. Why would universities be modeled with a society? The most notable features of society modding are the interface with its convenient display of a variable (society currency) and decisions ("powers"). What would you like to do with the currency of this society (these societies?) and what powers do you see them having?

**Comment by SEANSOKEN** [comment]
As stated currency is used to advance through the ranks. It wouldnt be a classical society where you just sit and wait and get random missions. Here you would also get random missions but also way more events for getting those points, since you are on the clock. The degree ranks shouldnt give you any power except the degree you get after finishing. Last two would have some powers and additions.
Also this society could be created as a dynamic so other places could form their own universities if they are rich enough. Like magic colleges can be formed.
At the end, it isnt about the powers, it is about the degree which will buff you for the rest of your life. Academic trait is already pretty nice to have. Now imagine being a graduated physician level 3.

As for full time job, well that is why landed rulers could get malus on advancing, since they need to study and manage realm. We could just give them an option to appoint a regent while they are at college, so their studying isnt getting any debuffs.

**Comment by HobbitDerrick** [comment]
OK. Without knowing it, you've gotten way out of hand on the amount of scripting you want here. Monastic society missions run to a bit over 2500 lines of script with moderately complex event chains, plus localization. That's more than all the Bretonnian events, more than five times the script for battle magic effects. If you want a more sophisticated and varied event chain for universities than the monastic missions (and it read like you did), that means weeks, probably months of work and testing just to get this subsystem going. Most modders - and I'm not talking just Gehe/Hex here, I mean CK2 modders - either avoid society missions or reuse vanilla ones because it's so much work just to get up to the amount of content that the vanilla missions have. So that quantity of content just for this request should be a non-starter.

But let's assume that we cut the quantity of work for this request down to something reasonable. What's the best way to realize it? The broad outline of your request is very similar to what I've just written for Chaos Dwarf Sorcerers. You can play around with that in the current Master if you want to check out the feel. Under the hood, characters acquire points - the exact mechanics are very hidden in the Chaos Dwarf case because Hashut's mind is beyond the feeble understanding of mortals - and once they have enough points, they "level up" and gain a trait. There are even some powers not strictly related to the trait-gaining path that spend that same currency, and I've earmarked some options for building onto the subsystem.

Now, I could have achieved that result by creating a Society, but I had a couple of reasons for not doing so. The cryptic, fickle God reason doesn't apply to universities, but that's an option of the no-society approach, rather than a requirement. (Currency values can be displayed in other ways.) I also wanted to be able to change how characters gained points depending on where they were in their path - and that probably applies to universities if you'd like courtiers to be able to study different subjects, switch majors, or even gain reduced benefits from their traits while distance-learning. Most importantly, I wanted courtiers to be able to pursue the daemonic power of Hashut (via the Sorcerer subsystem) and do crazy engineering things (via the Mad Inventors Society) at the same time. That concern does apply here because, for example, courtiers pursuing the blessing of Shallya (via the Order of the Bleeding Heart) and secular anatomical knowledge (via a university subsystem that isn't a society) sounds like something we'd want. To put that briefly: societies _only_ add the interface, and they come along with baggage that restricts scripting possibilities.

Bear with me while I talk about some of those other possibilities. Dynamically creating societies isn't a thing that really happens. I know some people pretend it does, but it doesn't - what they do is script a society, activate it during the game, and use dynamic localization to set the name to something context-sensitive. That means a little bit of work, but more importantly it sets a hard limit on the number of "dynamic" societies that can be created in a game. But dynamically creating titles is a thing that really happens and it doesn't matter much how many have already been created. (Adding more courts affects performance, of course.) So if you don't want a situation in which the University of Manaansufer can't be founded because universities already exist in Pigbarter, Remas, and Carlsbruck, societies are the wrong approach. And while societies don't bind you to using a single currency, they make that single currency that's displayed on the Society Interface much more prominent to the player - and that's a problem if you wanted to display (for example) grades in History, Philosophy, and Old World Theology all at once. Society currency has a nice hard-coded monthly tick, but you can realize it without societies too, and have options like giving AI courtiers a performance-friendly yearly tick while the player's currency updates more frequently (And intermediate options like player + immediate family using the frequent updates are doable too.). There's an optimization gadget for societies - but it gets less effective as more courtiers are assigned to societies. One could go on with examples, but let me switch to anecdote now. The AGoT team created pseudo-societies before MnM and resisted converting them into the societies enabled by that DLC for months because they knew societies were a less flexible route, before finally caving to player demands to make their mod worse. (Even then they kinda phoned it in and I don't blame them.)

In summary: the end-user side of your request, for more immersion and decisions with universities is totally reasonable, if a little off in terms of priorities. (The Empire is doing pretty well for immersion compared to the map average.) It's just that I think societies are a rotten, terrible way to realize those features that would handicap anyone bold enough to take on the scripting job as specified.

(p.s. This may come across as not just down on using society mechanics for universities, but generally down on adding societies. That really is where I stand. Players like societies so the mod should have significant society options in almost all regions, but once that requirement is fulfilled it's not a good idea to keep adding more just because we can.)

**Comment by SEANSOKEN** [comment]
We could talk over discord, you know
